### PR TITLE
VACMS-2457: Updates TableField caption help text.

### PR DESCRIPTION
**Pull request by ndouglas**
## Description

See #2457. 

## Testing done
Checked the field settings form for a couple of fields.

## Screenshots
Screen Shot 2020-11-09 at 2 30 28 PM

## QA steps
As an administrator:
- [ ] The settings form for [a TableField field on a given content type](http://va-gov-cms.lndo.site/admin/structure/types/manage/health_care_local_facility/fields/node.health_care_local_facility.field_facility_hours) should read `This brief caption will be associated with the table, rendered as a table heading in bold and will help screen reader better describe the content within.`

### CMS user-facing annoucement

Is an announcement needed to let editors know of this change? 
- [x] No announcement is needed for this code change. 
  - [x] Merge & carry on unburdened by announcements

### CODE REVIEW

File: docroot/modules/custom/va_gov_backend/va_gov_backend.module [hunk 1]
```diff
@@ -309,6 +309,16 @@ function va_gov_backend_form_alter(&$form, FormStateInterface $form_state, $form
 
 }
 
+/**
+ * Implements hook_form_FORM_ID_alter() for field_config_edit_form.
+ */
+function va_gov_backend_form_field_config_edit_form_alter(&$form, FormStateInterface $form_state, $form_id) {
+  $widget_type = $form['default_value']['widget'][0]['#type'] ?? NULL;
+  if ($widget_type === 'tablefield') {
+    $form['default_value']['widget'][0]['caption']['#description'] = 'This brief caption will be associated with the table, rendered as a table heading in bold and will help screen reader better describe the content within.';
```
Comment (acrollet) [inline docroot/modules/custom/va_gov_backend/va_gov_backend.module:318]: I believe we should use the translation API here, multi-lingual is coming at some point...

Comment (ndouglas) [reply docroot/modules/custom/va_gov_backend/va_gov_backend.module:318]: mmm, good point

Comment (ndouglas) [reply docroot/modules/custom/va_gov_backend/va_gov_backend.module:318]: that's something I've been lousy at tbh

**Review by swirtSJW — APPROVED** [review]

File: docroot/modules/custom/va_gov_backend/va_gov_backend.module [hunk 2]
```diff
@@ -309,6 +309,16 @@ function va_gov_backend_form_alter(&$form, FormStateInterface $form_state, $form
 
 }
 
+/**
+ * Implements hook_form_FORM_ID_alter() for field_config_edit_form.
+ */
+function va_gov_backend_form_field_config_edit_form_alter(&$form, FormStateInterface $form_state, $form_id) {
+  $widget_type = $form['default_value']['widget'][0]['#type'] ?? NULL;
+  if ($widget_type === 'tablefield') {
+    $form['default_value']['widget'][0]['caption']['#description'] = t('This brief caption will be associated with the table, rendered as a table heading in bold and will help screen reader better describe the content within.');
```
Comment (swirtSJW) [inline docroot/modules/custom/va_gov_backend/va_gov_backend.module:318]: Non-issue: This probably came in from UX but we should avoid describing the Front end result in too much detail as we are making promises that are not up to the CMS to keep ;)

Comment (ndouglas) [reply docroot/modules/custom/va_gov_backend/va_gov_backend.module:318]: Agreed.  I think this is probably a common (anti-)pattern with decoupled Drupal.

**Review by acrollet — APPROVED** [review]

**Pull request merged**